src/shaderlib/picking/shadertools-picking.md

```diff
@@ -0,0 +1,77 @@
+# picking (Shader Module)
+
+Provides support for color-coding-based picking and highlighting. In particular, supports picking a specific instance in an instanced draw call and highlighting an instance based on its picking color.
+
+Color based picking lets the application draw a primitive with a color that can later be used to index this specific primitive.
+
+Highlighting allows application to specify picking color of instance that need to be highlighted and the highlight color to be used.
+
+## Usage
+
+In your vertex shader, your inform the picking module what object we are currently rendering by supplying a picking color, perhaps from an attribute.
+```
+attribute vec3 aPickingColor;
+main() {
+  picking_setColor(aPickingColor);
+  ...
+}
+```
+
+In your fragment shader, you simply apply (call) the `picking_filterPickingColor` filter function at the very end of the shader. This will return the normal color, or the highlight color, or the picking color, as appropriate.
+```
+main() {
+  gl_FragColor = ...
+  gl_FragColor = picking_filterPickingColor(gl_FragColor);
+}
+```
+If you would like to apply the highlight color to the currently selected element call `picking_filterHighlightColor` before calling `picking_filterPickingColor`. You can also apply other filters on the non-picking color (vertex or highlight color) by placing those instruction between these two function calls.
+
+ ```
+main() {
+   gl_FragColor = picking_filterHighlightColor(color);
+    ... apply any filters on gl_FragColor ...
+  gl_FragColor = picking_filterPickingColor(gl_FragColor);
+}
+
+```
+
+## JavaScript Functions
+
+### getUniforms
+
+`getUniforms` takes an object takes a set of key/value pairs, returns an object with key/value pairs representing the uniforms that the `picking` module shaders need.
+
+`getUniforms(opts)`
+opts can contain following keys:
+* `pickingValid` (*boolean*) - When true current instance picking color is ignored, hence no instance is highlighted.
+* `pickingSelectedColor` (*array*) - Picking color of the currently selected instance.
+* `pickingHighlightColor` (*array*)- Color used to highlight the currently selected instance.
+* `pickingActive`=`false` (*boolean*) - Renders the picking colors instead of the normal colors. Normally only used with an off-screen framebuffer during picking. Default value is `false`.
+
+Note that the selected item will be rendered using `pickingHighlightColor`, if blending is enabled for the draw, alpha channel can be used to control the blending result.
```

Comment:
But what does that mean. We allow him to specify an RGBA value. We'll render it. And it will be blended like any other color.

What we should probably do is be very clear in the parameter documentation above that the `pickingHighlightColor` parameter expects/supports an RGBA color. In my view that would be sufficient.